src/gui/qgsbrowserdockwidget.h

```diff
@@ -102,6 +102,8 @@ class GUI_EXPORT QgsBrowserDockWidget : public QgsDockWidget, private Ui::QgsBro
     void openFile( const QString & );
     //! Emitted when drop uri list needs to be handled
     void handleDropUriList( const QgsMimeDataUtils::UriList & );
+    //! Connections changed in the browser
+    void connectionsChanged( );
```

Comment:
hi @elpaso - in general the new base class looks good to me - I have few suggestions though:
- can we switch from QDialog to QWidget as the base? this seems more appropriate when they are not used as standalone dialog anymore
- ```Qt::WindowFlags fl``` in the constructor is probably not necessary anymore (not used as standalone dialogs)
- ```WidgetMode``` enum could be probably moved from QgsProviderRegistry to this new class - maybe a more appropriate place?
- thinking about naming of the new class, how about something like ```QgsDataSourceWidget``` or ```QgsAbstractDataSourceWidget``` ? The name ```QgsSourceSelect``` does not say much...
- it would be nice to add a comment to ```connectionsChanged``` signal saying that it is used to refresh browser items

One more comment regarding the arcgis dialog - I would strongly prefer not to have it in public API - so I would suggest to either mark the class as private - or even better as Sandro suggested, to move it to arcgis provider where it belongs...